x-pack/test/pki_api_integration/apis/security/pki_auth.ts

```diff
@@ -0,0 +1,370 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import request, { Cookie } from 'request';
+import { delay } from 'bluebird';
+import { readFileSync } from 'fs';
+import { resolve } from 'path';
+// @ts-ignore
+import { CA_CERT_PATH } from '@kbn/dev-utils';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+const CA_CERT = readFileSync(CA_CERT_PATH);
+const FIRST_CLIENT_CERT = readFileSync(resolve(__dirname, '../../fixtures/first_client.p12'));
+const SECOND_CLIENT_CERT = readFileSync(resolve(__dirname, '../../fixtures/second_client.p12'));
+const UNTRUSTED_CLIENT_CERT = readFileSync(
+  resolve(__dirname, '../../fixtures/untrusted_client.p12')
+);
+
+export default function({ getService }: FtrProviderContext) {
+  const supertest = getService('supertestWithoutAuth');
+  const config = getService('config');
+
+  function checkCookieIsSet(cookie: Cookie) {
+    expect(cookie.value).to.not.be.empty();
+
+    expect(cookie.key).to.be('sid');
+    expect(cookie.path).to.be('/');
+    expect(cookie.httpOnly).to.be(true);
+    expect(cookie.maxAge).to.be(null);
+  }
+
+  function checkCookieIsCleared(cookie: Cookie) {
+    expect(cookie.value).to.be.empty();
+
+    expect(cookie.key).to.be('sid');
+    expect(cookie.path).to.be('/');
+    expect(cookie.httpOnly).to.be(true);
+    expect(cookie.maxAge).to.be(0);
+  }
+
+  describe('PKI authentication', () => {
```

Comment:
__note:__ the main problem in these tests is that we can't test that the subsequent request is authenticated via cookie and not via client certificate (as certificate should be always present and fingerprints should match). Not sure what to do here, I'm just ignoring this for now. We can potentially just query `.security-tokens` index and see if anything has changed there, but it feels like an overkill.

Reply:
That's a conundrum! Even if we used `clientAuthentication: optional`, we'd still trip on the logic which ensures that we present the certificate along with the cookie... We might end up having to rely on the unit tests for this.

Reply:
> We might end up having to rely on the unit tests for this.

Yeah, accepted that "internally" already :)